Proj/proj1_texasHoldem/C++/Proj2/DoxygenProject2/texasHoldemVer7/man/man3/Card.3
.TH "Card" 3 "Fri Jun 7 2019" "texasHoldem" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Card
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <Card\&.h>\fP
.PP
Inherited by \fBHands\fP\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBCard\fP ()"
.br
.ti -1c
.RI "\fBCard\fP (int f, int s)"
.br
.ti -1c
.RI "void \fBsetCard\fP (int f, int s)"
.br
.ti -1c
.RI "void \fBsetFace\fP (int f)"
.br
.ti -1c
.RI "void \fBsetSuit\fP (int s)"
.br
.ti -1c
.RI "void \fBsetACard\fP (\fBCard\fP &c)"
.br
.ti -1c
.RI "int \fBgetFace\fP () const"
.br
.ti -1c
.RI "int \fBgetSuit\fP () const"
.br
.ti -1c
.RI "void \fBdisplayCard\fP ()"
.br
.ti -1c
.RI "void \fBselectionSortByFace\fP (\fBCard\fP *c, int begin, int size)"
.br
.ti -1c
.RI "void \fBselectionSortBySuit\fP (\fBCard\fP *c, int begin, int size)"
.br
.ti -1c
.RI "const \fBCard\fP \fBoperator=\fP (const \fBCard\fP &right)"
.br
.in -1c
.SS "Protected Attributes"

.in +1c
.ti -1c
.RI "int \fBface\fP"
.br
.ti -1c
.RI "int \fBsuit\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 14 of file Card\&.h\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "Card::Card ()\fC [inline]\fP"

.PP
Definition at line 19 of file Card\&.h\&.
.SS "Card::Card (int f, int s)\fC [inline]\fP"

.PP
Definition at line 20 of file Card\&.h\&.
.SH "Member Function Documentation"
.PP 
.SS "void Card::displayCard ()"

.PP
Definition at line 13 of file Card\&.cpp\&.
.SS "int Card::getFace () const\fC [inline]\fP"

.PP
Definition at line 26 of file Card\&.h\&.
.SS "int Card::getSuit () const\fC [inline]\fP"

.PP
Definition at line 27 of file Card\&.h\&.
.SS "const \fBCard\fP Card::operator= (const \fBCard\fP & right)\fC [inline]\fP"

.PP
Definition at line 31 of file Card\&.h\&.
.SS "void Card::selectionSortByFace (\fBCard\fP * c, int begin, int size)"

.PP
Definition at line 34 of file Card\&.cpp\&.
.SS "void Card::selectionSortBySuit (\fBCard\fP * c, int begin, int size)"

.PP
Definition at line 51 of file Card\&.cpp\&.
.SS "void Card::setACard (\fBCard\fP & c)\fC [inline]\fP"

.PP
Definition at line 25 of file Card\&.h\&.
.SS "void Card::setCard (int f, int s)\fC [inline]\fP"

.PP
Definition at line 22 of file Card\&.h\&.
.SS "void Card::setFace (int f)\fC [inline]\fP"

.PP
Definition at line 23 of file Card\&.h\&.
.SS "void Card::setSuit (int s)\fC [inline]\fP"

.PP
Definition at line 24 of file Card\&.h\&.
.SH "Member Data Documentation"
.PP 
.SS "int Card::face\fC [protected]\fP"

.PP
Definition at line 16 of file Card\&.h\&.
.SS "int Card::suit\fC [protected]\fP"

.PP
Definition at line 17 of file Card\&.h\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for texasHoldem from the source code\&.


Proj/proj1_texasHoldem/texasHoldem_Ver1.1/Card.php
<?php

/*
 * File:   Card.php
 * Author: Byoung Mo Lee
 * Created on September 23, 2019 21:47 PM
 * Purpose: Texas Holdem
 * 
 */
define('MAXCRDS', 52);
define('MINCRDS', 0);

class Card {

    //put your code here
    private $number;
    private $nFace;  //2~J(11),Q(12),K(13),A(14)
    private $nSuit;  //0-Spades, 1-Hearts, 2-Diamonds, 3-Clubs
    private $sFaceName;
    private $sSuitName;
    private $sPict;

    public function Card($number) {
        if ($number >= MINCRDS && $number < MAXCRDS) {
            $this->number = $number;
            $this->setFace();
            $this->setSuit();
            $this->setPict();
        } else {
            $this->number = -1;
            $this->nFace = -1;
            $this->nSuit = -1;
            $this->sFaceName = "none";
            $this->sSuitName = "none";
            $this->sPict = "noen";
        }
    }

    public function setFace() {
        $num = $this->number % 13 + 1;

        if ($num == 1) {$num = 14;}
        $this->nFace = $num;
        
        
        switch ($this->nFace) {
            case 2: $this->sFaceName = "two";break;
            case 3: $this->sFaceName = "three";break;
            case 4: $this->sFaceName = "four";break;
            case 5: $this->sFaceName = "five";break;
            case 6: $this->sFaceName = "six";break;
            case 7: $this->sFaceName = "seven";break;
            case 8: $this->sFaceName = "eight";break;
            case 9: $this->sFaceName = "nine";break;
            case 10: $this->sFaceName = "ten";break;
            case 11: $this->sFaceName = "jack";break;
            case 12: $this->sFaceName = "queen";break;
            case 13: $this->sFaceName = "king";break;
            case 14: $this->sFaceName = "ace";break;
            default: $this->sFaceName = "Bad Value";
        }
    }

    public function setSuit() {
        if ($this->number < 13) {
            $this->nSuit = 0;
            $this->sSuitName = "spades";
        } else if ($this->number < 26) {
            $this->nSuit = 1;
            $this->sSuitName = "hearts";
        } else if ($this->number < 39) {
            $this->nSuit = 2;
            $this->sSuitName = "diamonds";
        } else if ($this->number < 52) {
            $this->nSuit = 3;
            $this->sSuitName = "clubs";
        } else {
            $this->suit = -1;
            $this->sSuitName = "Bad Value";
        }
    }

    
    private function setPict() {
        $this->sPict = "Cards/" . $this->sFaceName . $this->sSuitName.".jpg";
    }

    public function setACard(&$card) {
        $this->setCard($card->nFace, $card->nSuit);
    }

    public function getFace() {
        return $this->nFace;
    }

    public function getSuit() {
        return $this->nSuit;
    }

    public function getFaceName() {
        return $this->sFaceName;
    }

    public function getSuitName() {
        return $this->sSuitName;
    }

    public function getPict() {
        return $this->sPict;
    }

    public function toString() {
        echo "<img src=".$this->sPict." />";
        echo "<br/>Face Number= " . $this->nFace;
        echo "<br/>Suit Number= " . $this->nSuit;
        echo "<br/>Face Name= " . $this->sFaceName;
        echo "<br/>Suit Name= " . $this->sSuitName;
        echo "<br/><br/>";
    }
    
    public function displayCard(){
        echo "<img src=".$this->sPict." />";    
    }

}

?>
